workspaces/client/app/routes/dashboard/admin/equipments/internal/hooks/use-create-equipments.tsx

```diff
@@ -0,0 +1,40 @@
+import { useMutation, useQueryClient } from "@tanstack/react-query";
+import { useRepository } from "~/hooks/use-repository";
+import { useToast } from "~/hooks/use-toast";
+import type { EquipmentWithOwner } from "~/types/equipment";
+
+export const useCreateEquipment = () => {
+	const { equipmentRepository } = useRepository();
+	const queryClient = useQueryClient();
+	const { pushToast } = useToast();
+	return useMutation({
+		mutationFn: async (
+			equipment: Omit<
+				EquipmentWithOwner,
+				"id" | "createdAt" | "updatedAt" | "owner"
+			>,
+		) => {
+			await equipmentRepository.createEquipment({
+				name: equipment.name,
+				description: equipment.description ?? undefined,
+				ownerId: equipment.ownerId,
+			});
+			return equipment;
+		},
```

Comment:
もし https://github.com/saitamau-maximum/id/pull/271/files#r2268353886 がいけるのであれば return する必要がなくなるので、
```suggestion
		mutationFn: (equipment) => equipmentRepository.createEquipment(equipment),
```
このくらいシンプルになりそう (実際ほかの部分はこのくらいの量になっている)